### PR TITLE
issue/197 - Removed magic numbers in back-end and replaced with constants

### LINKED ISSUE #197: Remove Magic Numbers
- This isn't a specific task per se, it's more of an indication to create tasks related to this one.

### Acceptance Criteria:

1) Remove all magic numbers and replace with constants
2) Use enums (just JSON) where attributes can take one of a few set values, e.g. with theme colors

### Why is the feature required?

- This will prevent bugs due to changes in multiple places not being done
- It allows the code to be easier to read, as values can be traced back to an 'origin'

### PR DESCRIPTION
**Pull request by arangace**
-Removed magic numbers and replaced with more appropriate named constants.
-resolves #197
-Revisited from pull request #222  with feedback
- [ ] The pull request is complete according to the following criteria:
  - [x] Acceptance criteria have been met
  - [x] The documentation is kept up-to-date
  - [ ] Comprehensive tests (if applicable) have been generated and all pass.
  - [ ] The pull request describes the changes that have been made, and enough information is present in the description for any developer to understand what has changed
  - [ ] Commits have been squashed (or will be on merge).
  - [x] The branch name is descriptive and follows the pull request title format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [x] The pull request title is of the following format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [x] The description uses [github syntax](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to link to the issue. E,g Resolves se701g2/Doto#{Number}
  - [ ] At least two reviewers assigned. One of which must be the assigner of the issue.
  - [ ] If there are merge conflicts, run git rebase as opposed to git merge with master.

### CODE REVIEW

File: doto-backend/src/Constants/http-response.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    unauthorized: Number(401),
+    forbidden: Number(403),
+    successful: Number(200),
+    badRequest: Number(400),
+};
```
Comment (jordansimsmith) [inline doto-backend/src/Constants/http-response.js]: Any reason for the Number() syntax? Can we just have 

```
module.exports = { 
    unauthorized: 401
}
```

File: doto-backend/src/routes/task-route.js [hunk 2]
```diff
@@ -3,12 +3,12 @@ const router = express.Router();
 const authenticateToken = require("../config/token-setup").authenticateToken;
 const Task = require("../models/Task");
 const { logger } = require("../common/logging");
-
+const response = require("../Constants/http-response.js");
```
Comment (jordansimsmith) [inline doto-backend/src/routes/task-route.js]: When importing local files, usually the `.js` file extension is not used. See the above import for logging.

File: doto-backend/src/Constants/http-response.js (hunk 1 above)
Comment (FocalChord) [reply doto-backend/src/Constants/http-response.js:6]: I also feel like the keys should be capital because http-response.js is acting as an Enum object.

Comment (arangace) [reply doto-backend/src/Constants/http-response.js:6]: > Any reason for the Number() syntax? Can we just have
> 
> ```
> module.exports = { 
>     unauthorized: 401
> }
> ```

It wasn't working without it before so I looked up a solution which seemed to work if I had put the number syntax.

File: doto-backend/src/routes/task-route.js (hunk 2 above)
Comment (arangace) [reply doto-backend/src/routes/task-route.js:6]: > When importing local files, usually the `.js` file extension is not used. See the above import for logging.

Yup will fix that!

**Pull request merged**